android/modules/calendar/src/java/ti/modules/titanium/calendar/RecurrenceRuleProxy.java

```diff
@@ -0,0 +1,372 @@
+	}
+
+	//region Methods for converting native to Kroll values
+	private void fillFrequencyFields()
+	{
+		//reused variables in different cases
+		String days;
+		String byDay;
+		if (this.frequency != null) {
+			switch (this.frequency) {
+				case YEARLY:
+					Calendar cal = Calendar.getInstance();
+					cal.setTime(this.eventBegin);
+					//weeksOfTheYear
+					this.weeksOfTheYear = new int[] { cal.get(Calendar.WEEK_OF_YEAR) };
+					//monthsOfTheYear
+					this.monthsOfTheYear = new int[] { cal.get(Calendar.MONTH) };
+					days = matchExpression(".*(BYYEARDAY=[0-9]*).*", 10);
+					if (days != null) {
+						//daysOfTheYear
+						this.daysOfTheYear = new int[] { Integer.valueOf(days) };
+					}
+					break;
+				case MONTHLY:
+					//daysOfTheMonth
+					days = matchExpression(".*(BYMONTHDAY=(-)*[0-9]*).*", 11);
+					if (days != null) {
+						this.daysOfTheMonth = new int[] { Integer.valueOf(days) };
+					}
+					//daysOfTheWeek
+					byDay = matchExpression(".*(BYDAY=[,0-9A-Z]*).*", 6);
+					if (byDay != null) {
+						KrollDict daysOfTheWeekDictionary = new KrollDict();
+						daysOfTheWeekDictionary.put(this.dayOfWeekKey,
+													this.weekdaysMap.get(byDay.substring(byDay.length() - 2)));
+						daysOfTheWeekDictionary.put(this.weekNumberKey, byDay.substring(0, byDay.length() - 2));
+						this.daysOfTheWeek = new KrollDict[] { daysOfTheWeekDictionary };
+					}
+					break;
+				case WEEKLY:
+					//daysOfTheWeek
+					byDay = matchExpression(".*(BYDAY=[,0-9A-Z]*).*", 6);
+					// Split the days from result.
+					String[] daysArray = byDay.split(",");
+					this.daysOfTheWeek = new KrollDict[daysArray.length];
+					for (int i = 0; i < this.daysOfTheWeek.length; i++) {
+						KrollDict daysOfTheWeekDictionary = new KrollDict();
+						daysOfTheWeekDictionary.put(this.dayOfWeekKey, this.weekdaysMap.get(daysArray[i]));
+						// In the context of a weekly recurrence week number is irrelevant.
+						daysOfTheWeekDictionary.put(this.weekNumberKey, 0);
+						this.daysOfTheWeek[i] = daysOfTheWeekDictionary;
+					}
+					break;
+			}
+		}
+	}
+
+	private void calculateEnd()
+	{
+		// Check for until specific date condition.
+		String date = matchExpression(".*(UNTIL=[0-9A-Z]*).*", 6);
+		SimpleDateFormat sDateFormat = new SimpleDateFormat("yyyyMMdd");
+		if (date != null) {
+			try {
+				this.endDictionary.put(this.until, sDateFormat.parse(date.substring(0, 8)));
+			} catch (ParseException e) {
+				e.printStackTrace();
+			}
+		}
+
+		// Check for repeat count.
+		String until = matchExpression(".*(COUNT=[0-9]*).*", 6);
+		if (until != null) {
+			this.endDictionary.put(this.count, until);
+		}
+	}
+
+	private void calculateInterval()
+	{
+		String interval = matchExpression(".*(INTERVAL=[0-9]*).*", 9);
+		if (interval != null) {
+			this.interval = Integer.valueOf(interval);
+		}
+	}
+	//endregion
+
+	//region kroll proxy methods
+	@Kroll
+		.getProperty
+		@Kroll.method
+		public String getCalendarID()
```

Comment:
Unfortunately, adding multiple annotations to the same member triggers this clang formatting issue. We currently work-around it by adding the following comments around it like the below. (It's a bummer, I know. Sorry.)
```java
// clang-format off
@Kroll.getProperty
@Kroll.method
public String getCalendarID()
// clang-format on
{
	return this.calendarID;
}
```